### ISSUE TITLE
getting TypeError: <module '__main__'> is a built-in class

### ISSUE THREAD

**Issue by apdd2003** [flexxui/flexx#446]
getting type errors for all examples:
python=3.6
flexx=0.4.2

example1...
```
from flexx import app, ui
class Example(ui.HSplit):
    def init(self):
        ui.Widget(style='background:red;')
        ui.Widget(style='background:blue;')
>>
[I 21:07:16 flexx.app] Asset store collected 2 new modules.
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-cd79674d7e14> in <module>()
      1 from flexx import app, ui
      2 
----> 3 class Example(ui.HSplit):
      4 
      5     def init(self):

~/anaconda2/envs/py36/lib/python3.6/site-packages/flexx/event/_component.py in __init__(cls, name, bases, dct)
     52         cls._init_hook1(name, bases, dct)
     53         cls._set_summaries()
---> 54         cls._init_hook2(name, bases, dct)
     55         type.__init__(cls, name, bases, dct)
     56 

~/anaconda2/envs/py36/lib/python3.6/site-packages/flexx/app/_component2.py in _init_hook2(cls, cls_name, bases, dct)
    150 
    151         # Set JS on the JS class
--> 152         cls.JS.CODE = cls._get_js()
    153 
    154         # Register this class. The classes in this list will be automatically

~/anaconda2/envs/py36/lib/python3.6/site-packages/flexx/app/_component2.py in _get_js(cls)
    245 
    246         # Add this class
--> 247         c = create_js_component_class(cls.JS, cls_name, base_class_name)
    248         meta = c.meta
    249         code.append(c)

~/anaconda2/envs/py36/lib/python3.6/site-packages/flexx/event/_js.py in create_js_component_class(cls, cls_name, base_class)
    419 
    420     # Collect meta information of all code pieces that we collect
--> 421     mc = MetaCollector(cls)
    422     mc.meta['std_functions'].add('op_instantiate')  # b/c we use get_class_definition
    423 

~/anaconda2/envs/py36/lib/python3.6/site-packages/flexx/event/_js.py in __init__(self, cls)
     46             filename = getattr(sys.modules[cls.__module__], '__file__', None)
     47             linenr = (cls.__linenr__ if hasattr(cls, '__linenr__') else
---> 48                       inspect.findsource(cls)[1])
     49         self.meta = {'vars_unknown': set(), 'vars_global': set(),
     50                      'std_functions': set(), 'std_methods': set(),

~/anaconda2/envs/py36/lib/python3.6/inspect.py in findsource(object)
    766     is raised if the source code cannot be retrieved."""
    767 
--> 768     file = getsourcefile(object)
    769     if file:
    770         # Invalidate cache if needed.

~/anaconda2/envs/py36/lib/python3.6/inspect.py in getsourcefile(object)
    682     Return None if no way can be identified to get the source.
    683     """
--> 684     filename = getfile(object)
    685     all_bytecode_suffixes = importlib.machinery.DEBUG_BYTECODE_SUFFIXES[:]
    686     all_bytecode_suffixes += importlib.machinery.OPTIMIZED_BYTECODE_SUFFIXES[:]

~/anaconda2/envs/py36/lib/python3.6/inspect.py in getfile(object)
    652             if hasattr(object, '__file__'):
    653                 return object.__file__
--> 654         raise TypeError('{!r} is a built-in class'.format(object))
    655     if ismethod(object):
    656         object = object.__func__

TypeError: <module '__main__'> is a built-in class
```

example 2:
```
from flexx import app, ui
class Example(ui.Widget):
    def init(self):
        ui.Button(text='hello')
        ui.Button(text='world')

>>>
[I 21:10:19 flexx.app] Asset store collected 2 new modules.
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-fc038117ac31> in <module>()
      1 from flexx import app, ui
      2 
----> 3 class Example(ui.Widget):
      4 
      5     def init(self):

~/anaconda2/envs/py36/lib/python3.6/site-packages/flexx/event/_component.py in __init__(cls, name, bases, dct)
     52         cls._init_hook1(name, bases, dct)
     53         cls._set_summaries()
---> 54         cls._init_hook2(name, bases, dct)
     55         type.__init__(cls, name, bases, dct)
     56 

~/anaconda2/envs/py36/lib/python3.6/site-packages/flexx/app/_component2.py in _init_hook2(cls, cls_name, bases, dct)
    150 
    151         # Set JS on the JS class
--> 152         cls.JS.CODE = cls._get_js()
    153 
    154         # Register this class. The classes in this list will be automatically

~/anaconda2/envs/py36/lib/python3.6/site-packages/flexx/app/_component2.py in _get_js(cls)
    245 
    246         # Add this class
--> 247         c = create_js_component_class(cls.JS, cls_name, base_class_name)
    248         meta = c.meta
    249         code.append(c)

~/anaconda2/envs/py36/lib/python3.6/site-packages/flexx/event/_js.py in create_js_component_class(cls, cls_name, base_class)
    419 
    420     # Collect meta information of all code pieces that we collect
--> 421     mc = MetaCollector(cls)
    422     mc.meta['std_functions'].add('op_instantiate')  # b/c we use get_class_definition
    423 

~/anaconda2/envs/py36/lib/python3.6/site-packages/flexx/event/_js.py in __init__(self, cls)
     46             filename = getattr(sys.modules[cls.__module__], '__file__', None)
     47             linenr = (cls.__linenr__ if hasattr(cls, '__linenr__') else
---> 48                       inspect.findsource(cls)[1])
     49         self.meta = {'vars_unknown': set(), 'vars_global': set(),
     50                      'std_functions': set(), 'std_methods': set(),

~/anaconda2/envs/py36/lib/python3.6/inspect.py in findsource(object)
    766     is raised if the source code cannot be retrieved."""
    767 
--> 768     file = getsourcefile(object)
    769     if file:
    770         # Invalidate cache if needed.

~/anaconda2/envs/py36/lib/python3.6/inspect.py in getsourcefile(object)
    682     Return None if no way can be identified to get the source.
    683     """
--> 684     filename = getfile(object)
    685     all_bytecode_suffixes = importlib.machinery.DEBUG_BYTECODE_SUFFIXES[:]
    686     all_bytecode_suffixes += importlib.machinery.OPTIMIZED_BYTECODE_SUFFIXES[:]

~/anaconda2/envs/py36/lib/python3.6/inspect.py in getfile(object)
    652             if hasattr(object, '__file__'):
    653                 return object.__file__
--> 654         raise TypeError('{!r} is a built-in class'.format(object))
    655     if ismethod(object):
    656         object = object.__func__

TypeError: <module '__main__'> is a built-in class
```

**Comment by apdd2003** [comment]
edit: tried above examples in jupyter-notebook

**Comment by almarklein** [comment]
Are you using the IPython console or something?

You cannot define classes via the shell, because Python cannot find the source then. I suspect that this is going on, but I'd like to catch this error earlier so that Flexx can produce a more meaningful error message.

**Comment by apdd2003** [comment]
thanks for fast response... yes I was using jupyter-notebook

**Comment by almarklein** [comment]
> edit: tried above examples in jupyter-notebook

mmm, it should work in the notebook. Maybe something changed in a new version. Can you post the version of Python and IPython?

**Comment by apdd2003** [comment]
Python version: 3.6.4
Ipython: 6.2.1

**Comment by almarklein** [comment]
Thanks, I'll have a look soon.

**Comment by almarklein** [comment]
Fixed in #459
